AppCenterDataStorage/AppCenterDataStorage/Internal/Ingestion/MSCosmosDbIngestion.m

```diff
@@ -0,0 +1,48 @@
+#import "MSCosmosDbIngestion.h"
+#import "MSAppCenter.h"
+#import "MSAppCenterInternal.h"
+#import "MSHttpIngestionPrivate.h"
+#import "MSLoggerInternal.h"
+
+@implementation MSCosmosDbIngestion
+
+- (instancetype)init {
+  // @TODO: clean up once the MSHttpIngestion is refactored to take generic requests.
+  if ((self = [super initWithBaseUrl:@""
+                             apiPath:@""
+                             headers:@{}
+                        queryStrings:nil
+                        reachability:[MS_Reachability reachabilityForInternetConnection]
+                      retryIntervals:@[ @(10), @(5 * 60), @(20 * 60) ]])) {
+  }
+  return self;
+}
+
+- (NSURLRequest *)createRequest:(NSObject *)data eTag:(NSString *)__unused eTag {
+  NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:self.sendURL];
+
+  // Set method.
+  request.HTTPMethod = self.httpVerb;
+
+  // Set header params.
+  request.allHTTPHeaderFields = self.httpHeaders;
+
+  // Set body.
+  request.HTTPBody = (NSData *)data;
+
+  // Always disable cookies.
+  [request setHTTPShouldHandleCookies:NO];
+
+  // Log.
+  if ([MSLogger currentLogLevel] <= MSLogLevelVerbose) {
+    MSLogVerbose([MSAppCenter logTag], @"Creating CosmosDB request with URL: %@", request.URL.absoluteString);
+    MSLogVerbose([MSAppCenter logTag], @"CosmosDB request headers: %@", [super prettyPrintHeaders:request.allHTTPHeaderFields]);
+  }
+  return request;
+}
+
+- (NSString *)obfuscateHeaderValue:(NSString *)value forKey:(NSString *)key {
```

Comment:
There will be the same implementation in base class in identity feature branch. Add a TODO and remove (call parent method) implementation wouldn't forget to clean this up.
[PR](https://github.com/Microsoft/AppCenter-SDK-Apple/pull/1355/files#diff-f2a8eb528e7cb60df5c0546c8179dd91R99)